### PR TITLE
Improved BBOT Python API

### PR DESCRIPTION
**Pull request by TheTechromancer**
The following improvements have been made to BBOT's Python API in order to make it easier to use:

- A new "python" output module has been added which allows events to be consumed directly from the scanner (without using the `json` output module)
- `scan.start()` now yields BBOT event objects directly
- BBOT's scanner now defaults to the main BBOT config and will automatically convert the `config` argument to an OmegaConf object and merge it with the main config.

### CODE REVIEW

File: bbot/cli.py [hunk 1]
```diff
@@ -91,6 +91,9 @@ def main():
                                     log.verbose(f'Enabling {m} because it has flag "{f}"')
                                     modules.add(m)
 
+                if not options.output_modules:
+                    options.output_modules = ["human"]
```
Comment (kerrymilan) [inline bbot/cli.py:95]: How does this change interact with `bbot/scanner/scanner.py#60`? Which one takes precedence if no output module is specified?

File: bbot/scanner/scanner.py [hunk 2]
```diff
@@ -55,10 +57,12 @@ def __init__(
         if modules is None:
             modules = []
         if output_modules is None:
-            output_modules = ["human"]
+            output_modules = ["python"]
```
Comment (kerrymilan) [inline bbot/scanner/scanner.py:62]: Are events still printed to stdout when bbot defaults to using the python output module?

**Review by kerrymilan — COMMENTED** [review]
See comments in bbot/cli.py and bbot/scanner/scanner.py.

File: bbot/scanner/scanner.py (hunk 2 above)
Comment (TheTechromancer) [reply bbot/scanner/scanner.py:62]: No, they are yielded but not printed. Also this code should answer your first question. The python output module is default.

**Review by liquidsec — APPROVED** [review]

**Review by liquidsec — APPROVED** [review]

**Review by liquidsec — APPROVED** [review]

**Review by kerrymilan — APPROVED** [review]

**Pull request merged**